evap/results/views.py

```diff
@@ -102,14 +105,17 @@ def course_detail(request, semester_id, course_id):
             staff=request.user.is_staff)
     return render(request, "results_course_detail.html", template_data)
 
+def user_can_see_text_answer(user, course, text_answer):
+    if user == None:
+        return False
 
-def user_can_see_textresults(user, course, section):
-    if section.contributor == user:
-        return True
-    if course.is_user_responsible_or_delegate(user):
-        return True
-
-    if section.contributor in user.represented_users.all():
+    contributor = text_answer.contribution.contributor
+    if contributor == user:
         return True
+    if text_answer.published:
+        if course.is_user_responsible_or_delegate(user):
+            return True
+        if contributor in user.represented_users.all():
+            return True
```

Comment:
this and `text_answer.contribution.contributor` above smell like a whole bunch of database accesses. please check that you don't regress significantly here. maybe even add a test for that...? :)